.github/workflows/build_and_test.yml

```
@@ -0,0 +1,262 @@
+        if: always()
+
+  mining-tests:
+    needs: build
+    runs-on: ubuntu-latest
+    services:
+        bitcoind1:
+            image: ghcr.io/rsksmart/rskj/mit_bitcoind1:latest
+            ports:
+            - 8331:8331
+            - 31591:31591
+            - 32591:32591
+            options: --name bitcoind1
+        bitcoind2:
+            image: ghcr.io/rsksmart/rskj/mit_bitcoind2:latest
+            ports:
+            - 8332:8332
+            - 31592:31592
+            - 32592:32592
+            options: --name bitcoind2
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '12.x'
+      - name: Check Node.js version
+        run: node --version
+
+      - name: Checkout Mining Integration Tests Repository
+        uses: actions/checkout@v4
+        with:
+          repository: rsksmart/mining-integration-tests
+          ref: ${{ secrets.MINING_INTEGRATION_TESTS_REF }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: mining-integration-tests
+
+      - name: Install Mining Integration Tests dependencies
+        working-directory: mining-integration-tests
+        run: |
+          npm ci
+
+      - name: Change mining-integration-tests bitcoind url in config.json to localhost
+        working-directory: mining-integration-tests
+        run: |
+          sed -i 's/bitcoind01/localhost/g' config.json
+
+      - name: Generate BTC blocks
+        working-directory: mining-integration-tests
+        run: |
+          node --unhandled-rejections=strict generateBtcBlocks.js
+
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Verify Java installation
+        run: java -version
+
+      - uses: actions/cache@v4
+        name: Restore gradle cache
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+            rskj-core/build
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get Gradle jar
+        run: |
+          if [ ! -f ./gradle/wrapper/gradle-wrapper.jar ]; then
+            ./configure.sh
+          fi
+          
+      - name: Start RSKj and Run Tests
+        working-directory: mining-integration-tests
+        run: |
+          version=$(tr -d "'\"" < ../rskj-core/src/main/resources/version.properties \
+            | cut -d = -f 2- | paste -sd - -)
+          ls -la ../rskj-core/build/libs/rskj-core-"$version"-all.jar
+          java -Drsk.conf.file=./rsk-integration-test.conf -cp ../rskj-core/build/libs/rskj-core-"$version"-all.jar co.rsk.Start --regtest & rskpid=$!
+          until nc -z 127.0.0.1 4444
+            do
+              echo "Waiting for RskJ..."
+              sleep 1
+            done
```

The original block is over-indented. Besides that, this is "dangerous": if for any reason RskJ doesn't manage to be detected as up in port 4444, it'll loop forever. I suggest the following pattern instead:

```suggestion
          MAX_TRIES=30 tries=0
          while [ $tries -lt $MAX_TRIES ]; do
            sleep 1
            if nc -z 127.0.0.1 4444; then
              break
            fi
            tries=$(($tries + 1))
          done
          if [ $tries -eq $MAX_TRIES ]; then
            echo "Couldn't contact RskJ after $MAX_TRIES attempts." >&2
            exit 1
          fi
          done
```

---

I'm sure I had that there at some point. ah well, I'll put it back.